sdk/core/azure_core_amqp/src/fe2o3/messaging/mod.rs

```diff
@@ -211,114 +214,134 @@ impl
     ) -> Self {
         let mut amqp_message_builder = AmqpMessage::builder();
 
-        amqp_message_builder.with_body(AmqpMessageBody::Empty);
+        amqp_message_builder = amqp_message_builder.with_body(AmqpMessageBody::Empty);
 
         if let Some(application_properties) = message.application_properties {
-            amqp_message_builder.with_application_properties(application_properties.into());
+            amqp_message_builder =
+                amqp_message_builder.with_application_properties(application_properties.into());
         }
 
         if let Some(header) = message.header {
-            amqp_message_builder.with_header(header.into());
+            amqp_message_builder = amqp_message_builder.with_header(header.into());
         }
 
         if let Some(properties) = message.properties {
             info!("Converting properties to AmqpMessageProperties");
-            amqp_message_builder.with_properties(properties);
+            amqp_message_builder = amqp_message_builder.with_properties(properties);
         }
 
         if let Some(delivery_annotations) = message.delivery_annotations {
-            amqp_message_builder.with_delivery_annotations(delivery_annotations.0.into());
+            amqp_message_builder =
+                amqp_message_builder.with_delivery_annotations(delivery_annotations.0.into());
         }
 
         if let Some(message_annotations) = message.message_annotations {
-            amqp_message_builder.with_message_annotations(message_annotations.0.into());
+            amqp_message_builder =
+                amqp_message_builder.with_message_annotations(message_annotations.0.into());
         }
 
         if let Some(footer) = message.footer {
-            amqp_message_builder.with_footer(footer.0.into());
+            amqp_message_builder = amqp_message_builder.with_footer(footer.0.into());
         }
 
         amqp_message_builder.build()
     }
 }
 
+fn fe2o3_message_from_amqp_message(
+    message: &AmqpMessage,
+) -> fe2o3_amqp_types::messaging::Message<
+    fe2o3_amqp_types::messaging::Body<fe2o3_amqp_types::primitives::Value>,
+> {
+    let message_builder = fe2o3_amqp_types::messaging::Message::builder()
+        .application_properties(message.application_properties().map(|x| x.clone().into()))
```

Comment:
The `clone().into()` got me wondering - and this is non-blocking, but something to consider - if we're creating an owned message from a different message, isn't this whole outer function - and all those like it - really more of an `into()` itself? If so, couldn't it just take ownership and we avoid cloning altogether?

Reply:
It is - the .clone() is there because I don't have a From<&ApplicationProperties>. Which gets back to my earlier comment  about From<X> and From<&X>.

Let me think about the consequences of taking ownership here. In this case, application_properties returns a reference to the field within the structure, so it *can't* be a move operation, but there may be something else I can do here.